### PR TITLE
Add Statistics Variables to UserDefaults

### PR DESCRIPTION
**Pull request by jrweingart**

### CODE REVIEW

File: Breeze/UserDefaults.swift [hunk 1]
```diff
@@ -211,6 +229,100 @@ extension UserDefaults{
...
+    
+    func getProportionWeekSpent() -> Double {
+        //Current Date
+        let calendar = Calendar.current
+        let currDate = Date()
+        let dayComponents = calendar.dateComponents([.weekday], from: currDate)
+        let dayOfWeek = Int(dayComponents.weekday ?? 1)
+        let hourOfDay = calendar.component(.hour, from: currDate)
+        print("day: " + String(dayOfWeek))
+        print("hour: " + String(hourOfDay)) //HOW TO GET THIS TO MILITARY TIME
```
Comment (sab747) [inline Breeze/UserDefaults.swift:323]: I looked this up and found this thread: https://developer.apple.com/forums/thread/131724 , so I think we need to put this at the top of the method
let dateFormat = DateFormatter.dateFormat (fromTemplate: "j",options:0, locale: Locale.current)

File: Breeze/UserDefaults.swift [hunk 2]
```diff
@@ -211,6 +229,100 @@ extension UserDefaults{
...
+        let prevComponents = calendar.dateComponents([.weekday], from: prevDate)
+        let prevDayOfWeek = prevComponents.weekday
+        
+        let currDate = Date()
+        let currComponents = calendar.dateComponents([.weekday], from: currDate)
+        let currDayOfWeek = currComponents.weekday
+        
+        //If it is a new day, then reset the current phone usage and add it to statistics and send to Sendgrid
+        if (prevDayOfWeek != currDayOfWeek) {
+            resetCurrentPhoneUsage()
```
Comment (sab747) [inline Breeze/UserDefaults.swift:245]: maybe call addIntervalToCurrentPhoneUsage() so currentDayPhoneUsage updated before resetting currentPhoneUsage?

Comment (sab747) [reply Breeze/UserDefaults.swift:245]: Or actually we might need a check to see if protecteddata was available during the last check and then potentially also add the difference between time last check and midnight if the date has changed. and then add the remaining to the total for the new day.

**Review by sab747 — APPROVED** [review]
Awesome changes and additions! thorough and everything makes sense.

I think there is just one edge case when the date changes while the user is still on their phone but it also might already be handled, I have to look back at the checkPhoneUsage method.

File: Breeze/UserDefaults.swift (hunk 2 above)
Comment (sab747) [reply Breeze/UserDefaults.swift:245]: Agreed, I'm just hoping the intervals get small enough but either way tbh it shouldn't matter too much

Comment (jrweingart) [reply Breeze/UserDefaults.swift:245]: Actually I think what is going to happen is: let's say you are on your phone at 11:00pm one day, and then on your phone at 1:00am the next day, then currentPhoneUsage will reset, and then add the 2 hours. So the interval gets tagged onto the next day.

**Pull request merged**